### ISSUE TITLE
jobs: flake in TestRegistryLifecycle/rollback

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#52767]
Found in https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests/2178285?

```
------- Stdout: -------
=== RUN   TestRegistryLifecycle/rollback
--- FAIL: TestRegistryLifecycle/rollback (2.75s)
jobs_test.go:200: Starting resume
jobs_test.go:208: Exiting resume
jobs_test.go:200: Starting resume
jobs_test.go:208: Exiting resume
jobs_test.go:250: Starting success
jobs_test.go:254: Exiting success
jobs_test.go:200: Starting resume
jobs_test.go:648: expected job status: succeeded but got: running
jobs_test.go:208: Exiting resume
```

**Comment by knz** [comment]
I'll add a commit to skip the test for the time being.

**Comment by spaskob** [comment]
`make stress PKG=./pkg/jobs TESTFLAGS='-v ' TESTS=TestRegistryLifecycle/rollback`
does not reproduce the flake locally

**Comment by spaskob** [comment]
`make stress PKG=./pkg/jobs TESTFLAGS='-v ' TESTS=TestRegistryLifecycle`
also does not reproduce the flake locally in >2000 runs.

**Comment by knz** [comment]
Have you tried the git SHA where the issue was found originally.

It's possible the root cause has been  fixed on `master` already. If that is the case (i.e. you can repro in the past but not any more) it would be useful to determine which change was responsible.

**Comment by spaskob** [comment]
What would be that SHA? TC information is extremely hard to parse.

**Comment by knz** [comment]
> What would be that SHA? TC information is extremely hard to parse.

You always have two differnt ways to identify this:

1) identify the PR that the build is coming from. The PR number is at the top left 
![image](https://user-images.githubusercontent.com/642886/90607287-4fce8280-e201-11ea-9866-ba9be4c8e151.png)

   Then go to the PR, look at the history of the PR and find which build failed, and scan through the list of commit SHAs to find which one was used for the build.

2) go to the full test log like this:

   - Dependencies 
![image](https://user-images.githubusercontent.com/642886/90607444-81474e00-e201-11ea-97bc-cfea32b6502d.png)

   - Click the failed CI target, then click on the red failure message 
![image](https://user-images.githubusercontent.com/642886/90607506-97eda500-e201-11ea-9baa-22ffec3f6a70.png)

   - Click "download log" at the top right 
![image](https://user-images.githubusercontent.com/642886/90607543-a3d96700-e201-11ea-9b7a-13555aae7372.png)

   - The SHA is at the top  
![image](https://user-images.githubusercontent.com/642886/90607582-b358b000-e201-11ea-91ac-99994713e7b0.png)
